core/src/main/java/org/hisp/dhis/android/core/arch/db/adapters/custom/internal/DbVisualizationLegendColumnAdapter.kt

```diff
@@ -0,0 +1,79 @@
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  Redistributions of source code must retain the above copyright notice, this
+ *  list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *  this list of conditions and the following disclaimer in the documentation
+ *  and/or other materials provided with the distribution.
+ *  Neither the name of the HISP project nor the names of its contributors may
+ *  be used to endorse or promote products derived from this software without
+ *  specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.android.core.arch.db.adapters.custom.internal
+
+import android.content.ContentValues
+import android.database.Cursor
+import com.gabrielittner.auto.value.cursor.ColumnTypeAdapter
+import org.hisp.dhis.android.core.visualization.VisualizationLegend
+import org.hisp.dhis.android.core.visualization.VisualizationLegendSet
+import org.hisp.dhis.android.core.visualization.VisualizationTableInfo
+
+class DbVisualizationLegendColumnAdapter : ColumnTypeAdapter<VisualizationLegend> {
+
+    override fun fromCursor(cursor: Cursor, columnName: String): VisualizationLegend? {
+        val legendShowKeyColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_SHOW_KEY)
+        val legendStrategyColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_STRATEGY)
+        val legendStyleColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_STYLE)
+        val legendSetIdColumnIndex = cursor.getColumnIndex(VisualizationTableInfo.Columns.LEGEND_SET_ID)
+
+        val legendShowKey = cursor.getString(legendShowKeyColumnIndex)
+        val legendStrategy = cursor.getString(legendStrategyColumnIndex)
+        val legendStyle = cursor.getString(legendStyleColumnIndex)
+        val legendSetId = cursor.getString(legendSetIdColumnIndex)
+
+        val isLegendInfoPresent = listOf(legendShowKey, legendStrategy, legendStyle, legendSetId).all {
```

Comment:
LegendSetId might be null and still be a valid VisualizationLegend object. Actually, all the properties in VisualizationLegend are nullable, so we could build the object in any case

Reply:
It's means that the if check is not necessary ?

Reply:
We could remove the check, or replace the "all" by "any" so that if there is at least one property we build the object